protoc/flags.bzl

```diff
@@ -0,0 +1,25 @@
+PROTOC_TOOLCHAIN_FLAGS = {
+    "incompatible_enable_proto_toolchain_resolution": struct(
+        default = True,
+        description = """\
+        Bazel 7 introduced this flag to allow us fetch `protoc` rather than re-build it!
```

Comment:
We cannot use https://github.com/bazel-contrib/bazelrc-preset.bzl/blob/main/private/util.bzl. Maybe bazel_features can make them public so that we don't have to copy them in multiple projects

Reply:
Yeah we could encode the common functions as attributes like

```
struct(
  default = True,
  if_bazel_version_ge_same_major = "7.0.0",
)
```

which would work in nearly all cases. You might imagine allowing predicates to compose if you specify multiple (like "always AND") but probably don't even need it.

Reply:
I created https://github.com/bazel-contrib/bazelrc-preset.bzl/issues/67 to track this, but I don't think we need it for this PR, right?